### ISSUE TITLE
FWHM of exponentially modified gaussian is incorrect

### ISSUE THREAD

**Issue by cosama** [lmfit/lmfit-py#616]
#### Description

In https://github.com/lmfit/lmfit-py/blob/master/lmfit/models.py#L883 it is assumed that the FWHM of the Exponentially Modified Gaussian is calculated by `sigma*sqrt(8*log(2))`. This is not correct.

#### A Minimal, Complete, and Verifiable example
This code calculates the FWHM numerically, it shows that the above assumption is only true for gamma=1.
```
import numpy as np
import scipy

SQRT_TWO = np.sqrt(2) #1.414213562
FWHM_SIG_RATIO = np.sqrt(8*np.log(2))  # 2.35482

def expgauss(x, amp=1, mu=0, sigma=1.0, gamma=1.0):
    gss = gamma*sigma*sigma
    arg1 = gamma*(mu + gss/2.0 - x)
    arg2 = (mu + gss - x)/(SQRT_TWO*sigma)
    return amp*(gamma/2) * np.exp(arg1) * scipy.special.erfc(arg2)

def gauss(x, amp, mu, sigma):
    amp = np.abs(amp)
    mu = np.abs(mu)
    sigma = np.abs(sigma)
    return amp / np.sqrt(2. * sigma ** 2. * np.pi) * \
        np.exp(-(x - mu) ** 2. / (2. * sigma ** 2.))

def calc_fwhm(func, args, min=0, max=200):
    minimum = scipy.optimize.golden(lambda x: -func(x, *args))
    ma = func(minimum, *args)
    left = scipy.optimize.brentq(lambda x : func(x, *args) - ma*0.5,  min, minimum)
    right = scipy.optimize.brentq(lambda x : func(x, *args) - ma*0.5, minimum, max)
    return right - left

amp, mu, sigma, gamma = 10000, 100, 5, 0.25
for gamma in np.arange(0.05, 1.01, 0.05):
    fwhm = calc_fwhm(expgauss, args=(amp, mu, sigma, gamma))
    print(gamma, fwhm/sigma/FWHM_SIG_RATIO)
print("Gauss", calc_fwhm(gauss, args=(amp, mu, sigma))/sigma/FWHM_SIG_RATIO)
```
#### Error message:
Executing the above snipped produces the following output, which shows that the assumption for the FWHM is not correct for an Exponentially Modified Gaussian.

```
0.05 2.1337127622523266
0.1 1.5230313194815768
0.15000000000000002 1.3251520212658028
0.2 1.2276481141383135
0.25 1.1699380222211453
0.3 1.1321718208293392
0.35000000000000003 1.1058383583313516
0.4 1.0866493968715152
0.45 1.0722013571081739
0.5 1.0610416614777594
0.55 1.052242003193725
0.6000000000000001 1.0451829891025441
0.6500000000000001 1.0394369146895397
0.7000000000000001 1.0347000433937925
0.7500000000000001 1.0307515646891203
0.8 1.0274277229878135
0.8500000000000001 1.0246049651848192
0.9000000000000001 1.0221886531659001
0.9500000000000001 1.0201053207558615
1.0 1.018297250576952
Gauss 0.9999999999999994
```

I don't know if there is an analytic formula for this, however the implemented solution is wrong for sure.

**Comment by newville** [comment]
@cosama  If you have an analytic form, we'd be happy to use that. 

A quick look at the wikipedia link says the variance changes from `sigma**2` to `sigma**2 + 1/gamma**2`.   

Does it work better to then replace `sigma` by `sqrt(sigma**2 + 1/gamma**2)` so that 

    FWHM =   np.sqrt(8*np.log(2) * (sigma**2 + 1/gamma**2))

That seems like it must be a factor of sqrt(2) to big.   Anyway, I'd encourage you to check if that scales correctly for a good set of values for `sigma` and `gamma`.  If it gets to the right formula, fixing the code should be an easy: PR welcome, please guard against `gamma=0`!

**Comment by reneeotten** [comment]
Closed by d086e6205646fa6a2b6fad2b72e0a3a9e6f1134d @cosama: if you do happen to have an analytical formula do let us know!

**Comment by cosama** [comment]
I have not found an analytic formula, and could also not find a suitable empirical model. Removing them as per https://github.com/lmfit/lmfit-py/commit/d086e6205646fa6a2b6fad2b72e0a3a9e6f1134d is prudent.
